packages/application-shell-connectors/src/components/project-extension-image-regex/project-extension-image-regex.tsx

```diff
@@ -9,50 +10,52 @@ type TImageRegexOptions = {
   replace: string;
   search: string;
 };
-type TImageRegex = {
+export type TImageRegex = {
   thumb?: TImageRegexOptions;
   small?: TImageRegexOptions;
 };
 type TImageRegexContext = {
   isLoading: boolean;
-  imageRegex: TImageRegex | null;
+  imageRegex?: TImageRegex;
 };
 type ProviderProps = {
   children: React.ReactNode;
 };
 type ConsumerProps = {
-  render: (imageRegex: TImageRegexContext | {}) => React.ReactNode;
+  render: (imageRegex: TImageRegexContext) => React.ReactNode;
   children?: never;
 };
 
-const Context = React.createContext<TImageRegexContext | {}>({});
+const Context = React.createContext<TImageRegexContext>({ isLoading: true });
 
-const ProjectExtensionProviderForImageRegex = (props: ProviderProps) => (
-  <Query<
+const ProjectExtensionProviderForImageRegex = (props: ProviderProps) => {
+  const { loading, data } = useQuery<
     { projectExtension?: { imageRegex?: TImageRegex } },
-    { target: string }
-  >
-    query={FetchProjectExtensionImageRegex}
-    variables={{ target: GRAPHQL_TARGETS.SETTINGS_SERVICE }}
-  >
-    {({ loading, data }) => (
-      <Context.Provider
-        value={{
-          isLoading: loading,
-          imageRegex:
-            data && data.projectExtension && data.projectExtension.imageRegex,
-        }}
-      >
-        {props.children}
-      </Context.Provider>
-    )}
-  </Query>
-);
+    { target: typeof GRAPHQL_TARGETS.SETTINGS_SERVICE }
+  >(FetchProjectExtensionImageRegex, {
+    onError: reportErrorToSentry,
```

Comment:
Since we're not exposing any graphql error, we can use the `onError` handler to report it to sentry